Oleksandr.Svidovyi/Task_9/src/main/java/com/hillel/task_9/controller/ApiAccountController.java

```diff
@@ -0,0 +1,144 @@
+package com.hillel.task_9.controller;
+
+import com.hillel.task_9.model.AccountEntity;
+import com.hillel.task_9.model.ClientEntity;
+import com.hillel.task_9.model.request_model.TransferRequestModel;
+import com.hillel.task_9.model.request_model.UpdateAccountRequestModel;
+import com.hillel.task_9.service.AccountService;
+import com.hillel.task_9.service.ClientService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/clients/{id}/accounts")
+public class ApiAccountController {
+
+    @Autowired
+    private ClientService clientService;
+
+    @Autowired
+    @Qualifier("accountServiceImpl")
+    private AccountService accountService;
+
+
+    @GetMapping(path = "/list")
+    public ResponseEntity<List<AccountEntity>> getClientAccounts(@PathVariable int id) {
+        ClientEntity clientEntity = clientService.getClient(id);
+
+        if (clientEntity == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+
+        List<AccountEntity> accountEntities = clientService.getAccountsByClientId(id);
+
+        return new ResponseEntity<>(accountEntities, HttpStatus.OK);
+    }
+
+
+    @PostMapping
+    public ResponseEntity<ClientEntity> addAccount(@PathVariable int id, @RequestBody AccountEntity accountEntity) {
```

Comment:
Don't use AccountEntity as @RequestBody directly, 
Do it with CreateAccountRequest

Do the same for other methods in API part only

Reply:
Done